lib/client.c

```diff
@@ -1405,10 +1406,22 @@ SF_STATUS STDCALL snowflake_fetch(SF_STMT *sfstmt) {
                         }
                         break;
                     default:
-                        /* copy original data as is except Date/Time/Timestamp/Binary type */
+                        slen = strlen(raw_result->valuestring);
+                        log_debug("slen: %llu, maxbuflen: %llu, realloc func: %p",
+                                     slen,
+                                     result->max_length,
+                                     (void*)sfstmt->user_realloc_func);
+                        result->len = slen;
+                        if (sfstmt->user_realloc_func != NULL &&
+                            slen > result->max_length) {
```

Comment:
The buffer is reallocated ONLY IF:
- SF_STMT_USER_REALLOC_FUNC is set to the realloc function
- the buffer size is large enough to copy the data.